tests/test_graphql.py

```diff
@@ -256,3 +259,66 @@ async def test_query_environments_with_paging(server, client, setup_database):
     assert environments["pageInfo"]["endCursor"] == new_last_cursor
     assert environments["pageInfo"]["hasNextPage"] is True
     assert environments["pageInfo"]["hasPreviousPage"] is True
+
+
+async def test_is_environment_compiling(server, client, clienthelper, environment, mocked_compiler_service_block):
+
+    compilerslice = server.get_slice(SLICE_COMPILER)
+    assert isinstance(compilerslice, CompilerService)
+    env = await data.Environment.get_by_id(environment)
+
+    query = """
+    {
+        environments {
+            edges {
+                node {
+                  id
+                  isCompiling
+                }
+            }
+        }
+    }
+        """
+
+    def get_response(is_compiling: bool) -> dict:
+        return {
+            "data": {
+                "environments": {
+                    "edges": [
+                        {
+                            "node": {
+                                "id": environment,
+                                "isCompiling": is_compiling,
+                            }
+                        },
+                    ]
+                }
+            },
+            "errors": None,
+            "extensions": {},
+        }
+
+    result = await client.graphql(query=query)
+    assert result.code == 200
+    assert result.result["data"] == get_response(is_compiling=False)
+
+    # Trigger compile
+    await compilerslice.request_recompile(env=env, force_update=False, do_export=False, remote_id=uuid.uuid4())
+    # prevent race conditions where compile request is not yet in queue
+    await retry_limited(lambda: compilerslice._env_to_compile_task.get(uuid.UUID(environment), None) is not None, timeout=10)
+
+    # Assert that GraphQL reports that environment is compiling
+    result = await client.graphql(query=query)
+    assert result.code == 200
+    # Check if regular endpoint confirms that it is compiling
+    regular_check = await client.is_compiling(environment)
+    assert regular_check.code == 200
+    assert result.result["data"] == get_response(is_compiling=True)
+
+    # Finish compile
+    del compilerslice._env_to_compile_task[uuid.UUID(environment)]
```

Comment:
This is quite dirty. It relies on knowledge about the internals of the compiler service, which is bound to change and break the test case. Why not rely on `run_compile_and_wait_until_compile_is_done()` instead?